client/transactions/filters/index.tsx

```diff
@@ -8,45 +8,46 @@ import { getQuery } from '@woocommerce/navigation';
 /**
  * Internal dependencies
  */
-import { filters, advancedFilters, TransactionsFilterType } from './config';
+import { getFilters, getAdvancedFilters } from './config';
 import { formatCurrencyName } from '../../utils/currency';
 import './style.scss';
+import { __ } from '@wordpress/i18n/build-types';
 
 interface TransactionsFiltersProps {
 	storeCurrencies?: string[];
+	customerCurrencies?: string[];
 }
 
 export const TransactionsFilters = ( {
 	storeCurrencies,
+	customerCurrencies,
 }: TransactionsFiltersProps ): JSX.Element => {
-	const populateDepositCurrencies = (
-		filtersConfiguration: TransactionsFilterType[]
-	) => {
-		filtersConfiguration.forEach( ( filter ) => {
-			if ( 'store_currency_is' === filter.param ) {
-				const currencies = storeCurrencies || [];
-				// Generate select options: pick the first one (default) and add provided currencies
-				filter.filters = [
-					filter.filters[ 0 ],
-					...currencies.map( ( currencyCode: string ) => ( {
-						label: formatCurrencyName( currencyCode ),
-						value: currencyCode,
-					} ) ),
-				];
-				// Show the select when several currencies are available.
-				if ( 2 < filter.filters.length ) {
-					filter.showFilters = () => true;
-				}
-			}
-		} );
-		return filtersConfiguration;
-	};
+	const customerCurrencyOptions = customerCurrencies?.map(
+		( currencyCode: string ) => ( {
+			label: formatCurrencyName( currencyCode ),
+			value: currencyCode,
+		} )
+	);
+
+	const currencies = storeCurrencies || [];
+	const depositCurrencyOptions = currencies?.map(
+		( currencyCode: string ) => ( {
+			label: formatCurrencyName( currencyCode ),
+			value: currencyCode,
+		} )
+	);
```

Comment:
If we look at [transactions list definition for `storeCurrencies`](https://github.com/Automattic/woocommerce-payments/blob/7cb4bdaec540b6912d7d45041473bdd35c33e78b/client/transactions/list/index.tsx#L607) we're already providing an empty array as fallback value.

I think we could simplify `TransactionsFilters` code by making both props non-optional and just adding `|| []` for [`customerCurrencies` definition in transactions list.](https://github.com/Automattic/woocommerce-payments/blob/7cb4bdaec540b6912d7d45041473bdd35c33e78b/client/transactions/list/index.tsx#L608)

Reply:
Thanks, that simplifies the code nicely! I updated and also moved to using `useMemo` in 453d71d5c510c20e4a47881f4d23fa62c2fb0683